pkg/auxnetdevice/auxNetDevice_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auxnetdevice_test
+
+import (
+	"testing"
+
+	"github.com/jaypipes/ghw"
+	"github.com/jaypipes/pcidb"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/auxnetdevice"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/factory"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	tmocks "github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types/mocks"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils/mocks"
+)
+
+func newPciDevice(pciAddr string) *ghw.PCIDevice {
+	return &ghw.PCIDevice{
+		Address: pciAddr,
+		Vendor:  &pcidb.Vendor{ID: "Vendor"},
+		Product: &pcidb.Product{ID: "Product"},
+	}
+}
+
+func TestAuxnetdevice(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Auxnetdevice Suite")
+}
+
+var _ = Describe("AuxNetDevice", func() {
+	Describe("creating new AuxNetDevice", func() {
+		t := GinkgoT()
+		Context("successfully", func() {
+			It("should populate fields", func() {
+				fs := &utils.FakeFilesystem{
+					Dirs: []string{
+						"sys/bus/pci/devices/0000:00:00.1/net/net0",
+						"sys/bus/pci/drivers/foo",
+					},
+					Symlinks: map[string]string{
+						"sys/bus/pci/devices/0000:00:00.1/driver": "../../../../bus/pci/drivers/foo",
+					},
+					Files: map[string][]byte{"sys/bus/pci/devices/0000:00:00.1/numa_node": []byte("0")},
+				}
+				defer fs.Use()()
+				utils.SetDefaultMockNetlinkProvider()
+				fakeSriovnetProvider := mocks.SriovnetProvider{}
+				fakeSriovnetProvider.
+					On("GetUplinkRepresentorFromAux", "foo.bar.0").Return("net0", nil).
+					On("GetPfPciFromAux", "foo.bar.0").Return("0000:00:00.1", nil).
```

Comment:
can we have examples here (throughout this file) based on real data where applicable ?

Reply:
Ok